front/app/containers/SearchPage/components/ExportToCsvDialog.tsx

```diff
@@ -0,0 +1,70 @@
+import * as React from 'react';
+import gql from 'graphql-tag';
+import { graphql } from 'react-apollo';
+import { Modal } from 'react-bootstrap';
+import ThemedButton from 'components/StyledComponents/index';
+import ExportToCsvButton from './ExportToCsvButton';
+
+const SEARCH_EXPORT_QUERY = gql`
+  query SearchExportQuery($searchExportId: Int!) {
+    searchExport(searchExportId: $searchExportId) {
+      downloadUrl
+    }
+  }
+`;
+
+interface ExportToCsvDialogProps {
+  exportId?: Number;
+  data: any;
+  loading: boolean;
+  setExportId: any;
+}
+
+class ExportToCsvDialog extends React.Component<ExportToCsvDialogProps> {
+  componentWillMount() {
+    this.props.data.startPolling(250);
+  }
+
+  render() {
+    const { setExportId, loading, data } = this.props;
+
+    let body = <div />;
+    const noDownloadUrl =
+      data?.searchExport?.downloadUrl === null ||
+      data?.searchExport?.downloadUrl === undefined;
+    if (loading || noDownloadUrl) {
+      body = <span>Your export is running...</span>;
+    } else {
+      const openDownloadUrl = () => {
+        window.open(data.searchExport.downloadUrl);
+        data.stopPolling();
+        setExportId(null);
+      };
+      body = (
+        <ThemedButton onClick={openDownloadUrl}>Download Export</ThemedButton>
+      );
+    }
+
+    return (
+      <Modal show style={{ minHeight: '250px' }}>
```

Comment:
I was shocked to learn that this is the only place we're using a modal dialog in this app. I needed to add in some styles here and within the base css to get them to show reliably. It took me a while to figure out what was missing because we're on the older version of react-bootstrap and the documentation has never been great for it.